bundle/libraries/libraries.go

```diff
@@ -1,45 +1,77 @@
 package libraries
 
 import (
-	"context"
-	"fmt"
 	"path/filepath"
 
 	"github.com/databricks/cli/bundle"
-	"github.com/databricks/cli/bundle/config"
-	"github.com/databricks/cli/libs/cmdio"
-	"github.com/databricks/cli/libs/log"
-	"github.com/databricks/databricks-sdk-go/service/compute"
 	"github.com/databricks/databricks-sdk-go/service/jobs"
 )
 
-func findAllTasks(b *bundle.Bundle) []*jobs.Task {
+func findAllTasks(b *bundle.Bundle) map[string]([]*jobs.Task) {
 	r := b.Config.Resources
-	result := make([]*jobs.Task, 0)
+	result := make(map[string]([]*jobs.Task), 0)
 	for k := range b.Config.Resources.Jobs {
+		result[k] = make([]*jobs.Task, 0)
 		tasks := r.Jobs[k].JobSettings.Tasks
 		for i := range tasks {
 			task := &tasks[i]
-			result = append(result, task)
+			result[k] = append(result[k], task)
 		}
 	}
 
 	return result
 }
 
+func findAllEnvironments(b *bundle.Bundle) map[string]([]jobs.JobEnvironment) {
+	jobEnvs := make(map[string]([]jobs.JobEnvironment), 0)
+	for _, job := range b.Config.Resources.Jobs {
+		if len(job.Environments) == 0 {
+			continue
+		}
+
+		jobEnvs[job.Name] = job.Environments
+	}
+
+	return jobEnvs
+}
+
+func isEnvsWithLocalLibraries(envs []jobs.JobEnvironment) bool {
+	for _, e := range envs {
+		for _, l := range e.Spec.Dependencies {
+			if IsLocalPath(l) {
```

Comment:
What are accepted values for dependencies? If it is allowed to list pip-style dependencies, they may also be qualified as valid local paths using the current function.